### ISSUE TITLE
Disallow send | receive unless there's a snapshot - or similar

### ISSUE THREAD

**Issue by dm17** [openzfs/zfs#15785]
I've sent a few very large datasets without @snapshot without a problem in the past. Recently I realized this is actually not officially allowed. If a send|receive is interrupted zfs will give you a key to continue where you left off, but it won't work unless you were sending a snapshot - if you were sending a dataset without a snapshot specified, then you're out of luck. Also - and correct me if I'm wrong - if you send a dataset without a snapshot specified, then you cannot later send any snapshots onto that dataset.

I'm unsure what the correct behavior should be, but I do think it needs to be more obvious.

As an aside, it should obviously be more obvious which parameters for send|receive can be mixed. For example, it is not clear to me why a send fails without verbose information if I use -v or -P when trying to send --raw, but without -v & -P the send is allowed.

**Comment by mabod** [comment]
You can not send anything what is currently busy. snapshots are not busy (readonly) by default and can be send as is.  If you want to send a dataset, it needs to be unmounted. Only then you can send it. otherwise the erro message is: 

`warning: cannot send 'tank/test': target is busy; if a filesystem, it must not be mounted`

You can not continue a send of a dataset if the dataset was mounted and modified in the meantime.

**Comment by dm17** [comment]
Makes sense, but the dataset in question definitely wasn't busy as it was never imported on that machine. I don't see how a never-imported dataset could become busy (and there wasn't an active scrub on it during any sends).

**Comment by Haravikk** [comment]
A scrub or even resilver shouldn't cause it to become busy, as these work on unmounted datasets (neither of these cause updates to the dataset, they only fix what's already supposed to be there, so they wouldn't register as new for the purposes of send/receive).

Is there any chance the dataset had any of its properties changed? Changing properties on datasets can register as changes, you have to be careful with that when setting mount points and such on the receiving side to restore files (have to remember to rollback to the latest snapshot after you're done or you can't send to it until you do).
